### PR TITLE
Deprecate non RGB image conversion in rgb2gray

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

Following @hmaarrfk recommandations, this PR is a prerequisite to #4418.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [x] Unit tests
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by rfezzani** [comment]
Thank you @soupault !

**Comment by rfezzani** [comment]
Done! Thank you @jni ;-)

File: TODO.txt [hunk 1]
```diff
@@ -51,7 +51,8 @@ Version 0.19
 * Remove the warnings in ``skimage/filters/ridges.py`` from sato and hessian
   functions.
 * In ``skimage/color/colorconv.py``, remove  `rgb2grey` and `grey2rgb`.
-
+* In ``skimage/color/colorconv.py``, remove the deprecation warning
+  from `rgb2gray`.
```
Comment (hmaarrfk) [inline TODO.txt]:
```suggestion
  from `rgb2gray` pertaining to the check `rgb.shape[-1] > 3`.
```

**Comment by hmaarrfk** [comment]
Awesome!

**Comment by rfezzani** [comment]
Thank you @hmaarrfk , I did not have the time to commit your suggestion :-)

**Comment by hmaarrfk** [comment]
Yeah, I thought I could commit it for you. Sometimes I have the right to push to your repo. I would rather you focus on the other PRs you have flying around ;)

**Pull request merged**